### PR TITLE
[FE] React router and home page styles

### PR DESCRIPTION
**Pull request by ksittner**
- Added routing to Home and its subcomponents
- Added styling to Home and its subcomponents
- InteractBar now properly updates Map when boxes are checked
- Fixed focus issue with input in SearchBar component
- Organized the /src directory in the frontend into /components, /icons, and /styles

### CODE REVIEW

**Comment by davidayu** [comment]
The changes look good to me!

**Review by davidayu — APPROVED** [review]

**Comment by AmeyaMalekar** [comment]
Looks good Kevin.

**Comment by noahmarxx** [comment]
Nice job Kevin.

File: backend/backend.js [hunk 1]
```diff
@@ -30,8 +30,8 @@ app.get("/pins", async (req, res) => {
     res.send(result);
   } else if (pinType != undefined) {
     let result = await pinServices.filterByType(pinType);
-    if (result === undefined || result === null)
-      res.status(404).send("Resource not found.");
+    result = { pins_list: result };
```
Comment (eliasow) [inline backend/backend.js]: I think this might work better if placed after if(!result) check

File: backend/backend.js [hunk 2]
```diff
@@ -30,9 +30,10 @@ app.get("/pins", async (req, res) => {
     res.send(result);
   } else if (pinType != undefined) {
     let result = await pinServices.filterByType(pinType);
-    if (result === undefined || result === null)
+    if (!result) {
```
Comment (eliasow) [inline backend/backend.js:33]: Thank you for making logic order change, Looks good to me

**Review by eliasow — COMMENTED** [review]
Looks Good

**Comment by ksittner** [comment]
> The changes look good to me!



> The changes look good to me!

Thank you for the review

**Comment by ksittner** [comment]
> The changes look good to me!

Thank you for the review

**Comment by ksittner** [comment]
> Looks good Kevin.

Thank you Ameyah

**Comment by ksittner** [comment]
> Nice job Kevin.

Thanks for reviewing, Noah

File: backend/backend.js (hunk 1 above)
Comment (ksittner) [reply backend/backend.js]: > 

Good catch, this should be resolved now

**Pull request merged**